### PR TITLE
Added access to ble_scan_result_evt_param as get_scan_result

### PR DESCRIPTION
**Pull request by myhomeiot**
# What does this implement/fix? 

This will allow to create ESPBTDeviceListener's which will have full access to scan result and allow to parse RAW data or reconstruct BLE Advertising packet and forward it to external processing like this done in [BLE Gateway](https://github.com/myhomeiot/esphome-components#ble-gateway) component [here](https://github.com/myhomeiot/esphome-components/blob/83aae9078215130b3e17bee360c10268fc890059/components/ble_gateway/ble_gateway.cpp#L16).

Thanks in advance!

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by andrewjswan** [comment]
Great improvement, hope for a soon release as part of ESPHome.

**Review by oxan — APPROVED** [review]

File: esphome/components/esp32_ble_tracker/esp32_ble_tracker.h [hunk 1]
```diff
@@ -121,6 +123,7 @@ class ESPBTDevice {
   std::vector<ESPBTUUID> service_uuids_;
   std::vector<ServiceData> manufacturer_datas_{};
   std::vector<ServiceData> service_datas_{};
+  esp_ble_gap_cb_param_t::ble_scan_result_evt_param scan_rst_{};
```
Comment (jesserockz) [inline esphome/components/esp32_ble_tracker/esp32_ble_tracker.h]: `rst` is confusing, lets just name it `result`
```suggestion
  esp_ble_gap_cb_param_t::ble_scan_result_evt_param scan_result_{};
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/esp32_ble_tracker/esp32_ble_tracker.h (hunk 1 above)
Comment (myhomeiot) [reply esphome/components/esp32_ble_tracker/esp32_ble_tracker.h]: Fixed.
PS: Previously I did it to fit existing parse_scan_rst.

Comment (jesserockz) [reply esphome/components/esp32_ble_tracker/esp32_ble_tracker.h]: Ah ok. I will change that too. Thanks

**Review by jesserockz — APPROVED** [review]

**Pull request merged**